src/libraries/Microsoft.Extensions.Logging/src/Logger.cs

```diff
@@ -150,15 +152,102 @@ static bool LoggerIsEnabled(LogLevel logLevel, ILogger logger, ref List<Exceptio
+        {
+            throw new AggregateException(
+                message: "An error occurred while writing to logger(s).", innerExceptions: exceptions);
+        }
+
         internal string DebuggerToString()
         {
             return DebuggerDisplayFormatting.DebuggerToString(_categoryName, this);
         }
 
-        private static void ThrowLoggingError(List<Exception> exceptions)
+        private sealed class LoggerDebugView(Logger logger)
         {
-            throw new AggregateException(
-                message: "An error occurred while writing to logger(s).", innerExceptions: exceptions);
+            public string Name => logger._categoryName;
+
+            // The list of providers includes the full list of configured providers from the logger factory.
+            // It then mentions the min level and enable status of each provider for this logger.
+            public List<LoggerProviderDebugView> Providers
+            {
+                get
+                {
+                    List<LoggerProviderDebugView> providers = new List<LoggerProviderDebugView>();
+                    for (int i = 0; i < logger.Loggers.Length; i++)
+                    {
+                        LoggerInformation loggerInfo = logger.Loggers[i];
+                        string providerName = ProviderAliasUtilities.GetAlias(loggerInfo.ProviderType) ?? loggerInfo.ProviderType.Name;
+                        MessageLogger? messageLogger = logger.MessageLoggers?.FirstOrDefault(messageLogger => messageLogger.Logger == loggerInfo.Logger);
+
+                        providers.Add(new LoggerProviderDebugView(providerName, messageLogger));
+                    }
+
+                    return providers;
+                }
+            }
+            public List<object?>? Scopes
+            {
+                get
+                {
+                    var scopeProvider = logger.ScopeLoggers?.FirstOrDefault().ExternalScopeProvider;
+                    if (scopeProvider == null)
+                    {
+                        return null;
+                    }
+
+                    List<object?> scopes = new List<object?>();
+                    scopeProvider.ForEachScope((scope, scopes) => scopes!.Add(scope), scopes);
+                    return scopes;
+                }
+            }
+            public LogLevel? MinLevel => DebuggerDisplayFormatting.CalculateEnabledLogLevel(logger);
+            public bool Enabled => DebuggerDisplayFormatting.CalculateEnabledLogLevel(logger) != null;
+        }
+
+        [DebuggerDisplay("{DebuggerToString(),nq}")]
+        private sealed class LoggerProviderDebugView(string providerName, MessageLogger? messageLogger)
+        {
+            public string Name => providerName;
+            public LogLevel LogLevel => CalculateEnabledLogLevel(messageLogger) ?? LogLevel.None;
+
+            private static LogLevel? CalculateEnabledLogLevel(MessageLogger? logger)
```

Comment:
>CalculateEnabledLogLevel [](http://example.com/codeflow?start=37&length=24)

Can't this uses `DebuggerDisplayFormatting.CalculateEnabledLogLevel` or vice versa? Just to avoid duplicating the code.

Reply:
No. ILogger vs MessageLogger arg.